cmdutil/debug/debug.go

```diff
@@ -70,4 +83,69 @@ func (s *Server) Stop(_ error) {
 	agent.Close()
 
 	close(s.done)
+
+	if s.pprofServer != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		if err := s.pprofServer.Shutdown(ctx); err != nil {
+			s.logger.WithError(err).Error("Error shutting down pprof server")
+		}
+	}
+}
+
+// ProfileConfig holds the configuration for the pprof server.
+type ProfileConfig struct {
+	Addr            string
+	ProfileHandlers map[string]http.HandlerFunc
+}
+
+// NewPprofServer sets up a pprof server with configurable profiling types and returns a Server instance.
+func NewPprofServer(config ProfileConfig, l logrus.FieldLogger) *Server {
+	if config.Addr == "" {
+		config.Addr = "127.0.0.1:9998" // Default port
+	}
+
+	// Create a new HTTP mux for handling pprof routes.
+	mux := http.NewServeMux()
+
+	// Iterate over the profile handlers and add them to the mux.
+	for profile, handler := range config.ProfileHandlers {
+		if handler != nil {
+			if profile == "mutex" {
+				runtime.SetMutexProfileFraction(2)
```

Comment:
I think this should be configurable on the profile config.

Reply:
I mean something like this:
```
type PProfServerConfig struct {
  MutextProfileFraction int
}
```
If set to zero _and_ the mutex profile is enabled, we can use a reasonable default of 2.